### ISSUE TITLE
Air 33 and Android 64 Bits

### ISSUE THREAD

**Issue by rewb0rn** [adjust/adobe_air_sdk#49]
Hi,

I hope all goes well! As you may have heard, HARMAN took over the future development of the Adobe Air SDK, including the upcoming requirement to release 64 Bits apps on Google Play from the beginning of August.

My understanding is that all ANEs have to be repackaged to support 64 Bits on Android.

The release for Air SDK 33 beta that includes 64 Bits on Android is announced for this month. It would be great if you could get in touch with HARMAN at [Adobe.Support @ harman.com] to make sure you can get a preview of the Air 33 SDK and release an updated 64 Bits ANE in July.

Thanks in advance!

**Comment by uerceg** [comment]
Hi @rewb0rn 

And in general, thank you very much for keeping us up to date with all bunch of various topics from Adobe AIR world. 🍺 

Thanks for advice, we'll write to them and see if we can get in possession of the AIR SDK 33 beta to rebuild our ANE with it. Will keep you posted on the topic.

Cheers!

**Comment by jeuxdaily** [comment]
hello

it's the good news
thanks man

**Comment by uerceg** [comment]
Update: We reached out, hopefully we'll get their 33 beta SDK in some form to give it a shot.

**Comment by rewb0rn** [comment]
Hi, any news on this?

**Comment by uerceg** [comment]
Hey @rewb0rn,

I can just say yes, but without much of a context. I was following emails for a while, but at that time (a bit ago) SDK was still not ready, but I'd say that maybe they have something we can work with at this moment, need to check those last emails. We were having more urgent issues on our side, but hopefully I'll be able to devote some time to this topic beginning next week.

**Comment by rewb0rn** [comment]
Okay great. Just a bit more of info: A beta SDK is ready and available to anyone that writes to Harman. We also have a version of it here and are testing our apps with it.

**Comment by rewb0rn** [comment]
Hello, any news about this? The time has come for us to update our apps :)

**Comment by uerceg** [comment]
Ehey @rewb0rn,

We cleared up all doubts we had on legal side with updated T&Cs of new SDK so we're good to proceed with this now. Not sure if you're following mails which Andrew is writing in that mail list, but seems that Google has postponed necessity for apps to be updated to support 64-bits for until 2020, so seems like not super high pressing thing, right? Not that I'm saying this to postpone our SDK update even more 😄, we'll get to it definitely, but seems like there's no strict deadline in 2019 for that anymore.

**Comment by rewb0rn** [comment]
Yes correct, we do follow the mailing list. However the postponing is only valid for existing apps. New apps still must submit a 64 bits build and we plan to release a new app soon. In the meantime, we would like to release 64 bits versions of our apps anyway, since we it should bring performance benefits.

**Comment by uerceg** [comment]
Ah, good to know. 👍 We should start working on this topic this week and will keep you posted.

**Comment by jeuxdaily** [comment]
5 month without news from HARMAN. so what we can do with our apps in google play or the future apps

**Comment by jeuxdaily** [comment]
I thanks that :

There is not, never was, never will be support for a 64-bit development/test environment from Adobe. The Adobe toolchain for AIR/Flex development is entirely based on 32-bit java development tools [primarily in the jar files found in the bin and lib directories of the SDK releases]

**Comment by uerceg** [comment]
TBH, not sure what to say on this quote of yours. Seems like you found it in here: http://www.flashdevelop.org/community/viewtopic.php?f=9&t=13065. Also seems like a year old. 

From what I managed to follow on this topic in months behind us, Harman does have Adobe AIR SDK 33 ready to be used which will contain 64-bit support needed for Android apps that are supposed to be newly published this year or updated by 2nd half of 2020.

**Comment by jeuxdaily** [comment]
that's mean no update and no 64-bit support by air sdk 

if that is true

by by adobe!!!!!!

**Comment by rewb0rn** [comment]
Adobe Air 33 SDK is available, we have built a 64 Bits version of our games yesterday, so please don't spread misinformation. You need to contact Harman if you want to have access to it.

Best regards

**Comment by uerceg** [comment]
@rewb0rn I have pinged Andrew since a while ago, but I still didn't get any answer. Do you know where can I download Adobe AIR SDK 33? I have tried finding download link somewhere on their website, but no luck. Is it somewhere publicly available or do I need to ping them in order to get it?

**Comment by rewb0rn** [comment]
Right now there is no direct way to download it until their website is live. You need to contact them directly via email to Adobe.Support@harman.com

**Comment by uerceg** [comment]
Thanks for you answer, did that on both, that and Andrew's email address. Still no response, so once we get in possession of SDK, we'll start working our ANE update ASAP. Until then, :loading:.

**Comment by rewb0rn** [comment]
Thanks, I have written them an email to check on that.

**Comment by rewb0rn** [comment]
Got a response, they had someone from the Adjust team on their mailing list and have send out the SDK license agreements as part of the normal mailshots, but have not had any response as far as they can tell.

Can you double check if it went to your spam?

**Comment by uerceg** [comment]
Ehey, didn't wanna update in here until fully settled, but thank you very much for pinging them! They reached out and discussion is ongoing, soon we should be able to get the SDK and they even gave us instructions how should ANE be updated, super helpful. Thanks one more time!

As soon as we have updated ANE version, is it okay if I attach it in here for you to double check it if it's built properly and if yes, from there, we'll make an official release?

**Comment by rewb0rn** [comment]
Sure, I will be happy to give it a test before you release it officially!

**Comment by jeuxdaily** [comment]
hello friends
I have received this email

> 
Hi

 

Forwarding for those who have registered their interest recently – please see below.

 

The SDK is now available, so if you would like to receive it, please confirm your acceptance of the attached documents which describe the Ts & Cs and the pricing for this. Please also let us know which pricing tier you/your company would fall into; we’re still finalising the payment process although it looks like we can already receive funds via PayPal; we’re hoping to get an online payment mechanism set up this month.

 

many thanks

 

   Andrew

> 

thats mean dsk 33 is avaliable

**Comment by jeuxdaily** [comment]
https://blog.expo.io/@esamelson

**Comment by uerceg** [comment]
Okay, let's give it a shot. @rewb0rn Here's our ANE updated in accordance to Harman's instructions on how to support Android 64-bit and compiled with Adobe AIR SDK v33. Looking forward to hear from you does it work for you as expected.

[Adjust-4.18.0.ane.zip](https://github.com/adjust/adobe_air_sdk/files/3501465/Adjust-4.18.0.ane.zip)

**Comment by jeuxdaily** [comment]
hello

please link of AIR SDK 33

**Comment by uerceg** [comment]
@jeuxdaily 

You need to write to Harman support and ask for AIR SDK v33 from them. You need to accept T&C document they will send you to check and after that, they will send you the SDK. You probably need to agree with them which package you belong to and how much you need to pay for it if you are planning to publish commercial apps built with that SDK to stores.

Cheers

**Comment by jeuxdaily** [comment]
I do
I'm waiting
but If you have it send me link

**Comment by uerceg** [comment]
There is no link. It's not publicly available for download. You need to get download link from Harman.

**Comment by jeuxdaily** [comment]
why not

so this SDK not free

**Comment by uerceg** [comment]
Well, it's written in email quote you shared with us in [here](https://github.com/adjust/adobe_air_sdk/issues/49#issuecomment-521040491).

```
Please also let us know which pricing tier you/your company would fall into; we’re still finalising the payment process although it looks like we can already receive funds via PayPal; we’re hoping to get an online payment mechanism set up this month.
```

**Comment by jeuxdaily** [comment]
I'm individuel  i don't have company
thanks for your info

**Comment by rewb0rn** [comment]
@uerceg Thanks for the build! First tests look good, I can see the 64 bits build show up in the sandbox console and events are being tracked as well. We will try to verify trackers on monday. Anything we should focus on in particular? Did you test all of the features on your side?

**Comment by uerceg** [comment]
@rewb0rn Thanks for the update, good news. No really. This upcoming version contains some under the hood stuff which should not affect any of the existing features at all. All of them are tested on our side with this 4.18.0 build I sent you and is looking fine on our side. Of course double check if everything works for your use cases on your side just in case.

So from functionality point of view everything should work as expected. We would just need info from you does this newly packaged ANE works in your 64-bit supported app as expected to make sure we properly added support for 64-bit.

**Comment by rewb0rn** [comment]
Yes, I am testing the 64 bits build right now and I can confirm that at least on the basic level this is working as expected. I will ping you in case we find any problems, but I can confirm that on the first test, 64 bits is working just fine :)

**Comment by jeuxdaily** [comment]
hello

please see this problem

https://github.com/Gamua/Adobe-Runtime-Support/issues/76

**Comment by uerceg** [comment]
@jeuxdaily We are not really here to solve Adobe AIR issues outside of those which might be caused by Adjust ANE.

I don't even know what is the issue you're facing based on your description in there. Sounds to me like namespace version issue in XML configuration file, but no idea.

If you are using Adobe AIR SDK v33, you should declare the namespace version like this:

```xml
<application xmlns="http://ns.adobe.com/air/application/32.0"> 
```

**Comment by jeuxdaily** [comment]
this my apps-app.xml

<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<application xmlns="http://ns.adobe.com/air/application/32.0">
    <id>com.ccicggdfttsdklmsrdddrtfmhd.dfccgtffderrt</id>
    <versionNumber>1.0.0</versionNumber>
    <versionLabel></versionLabel>
    <filename>cake games cooking delicious for girls</filename>
	<architecture>64</architecture>
    <description/>
 
  
    <name>first apps</name>


  
    <copyright/>
    <initialWindow>
        <content>first-apps.swf</content>
        <systemChrome>standard</systemChrome>
        <transparent>false</transparent>
        <visible>true</visible>
        <fullScreen>true</fullScreen>
        <aspectRatio>landscape</aspectRatio>
        <renderMode>direct</renderMode>
        <autoOrients>false</autoOrients>
    </initialWindow>

    <icon>
        
        
        
        <image96x96>icons/96.png</image96x96>
        <image144x144>icons/144.png</image144x144>
        <image192x192>icons/192.png</image192x192>
    <image36x36>icons/36.png</image36x36>
    <image48x48>icons/48.png</image48x48>
    <image72x72>icons/72.png</image72x72>
  </icon>
    <customUpdateUI>false</customUpdateUI>
    <allowBrowserInvocation>false</allowBrowserInvocation>
    <android>
        <manifestAdditions><![CDATA[<manifest android:installLocation="auto">
	  <uses-sdk android:minSdkVersion="9" android:targetSdkVersion="32" />
			<uses-permission android:name="android.permission.INTERNET"/>
			<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
			<uses-permission android:name="com.samsung.android.providers.context.permission.WRITE_USE_APP_FEATURE_SURVEY"/>
			<uses-permission android:name="android.permission.REQUEST_INSTALL_PACKAGES"/>
			<application>
						
						<meta-data android:name="com.google.android.gms.version" android:value="@integer/google_play_services_version"/>
						
						 <activity android:name="com.google.android.gms.ads.AdActivity" android:configChanges="keyboard|keyboardHidden|orientation|screenLayout|uiMode|screenSize|smallestScreenSize" android:theme="@android:style/Theme.Translucent" />
			</application>
		</manifest>]]></manifestAdditions>
    </android>
    <supportedLanguages>en zh cs nl fr de it ja ko pl pt ru es sv tr</supportedLanguages>

  <extensions>
    <extensionID>com.milkmangames.extensions.AdMob</extensionID>
    <extensionID>com.milkmangames.extensions.GoogleServices</extensionID>
  </extensions>
</application>

**Comment by jeuxdaily** [comment]
I change 32 to 33 but when i compile my app tp apk this change doesn't work

**Comment by uerceg** [comment]
Honestly no idea. Maybe you will manage to get reply in the repo where you asked the issue. Good luck.

And please in here stick to the topic of the issue, any further comment unrelated to the topic of adding 64-bit support to Adjust ANE will be deleted.

Thank you for your understanding.

**Comment by jeuxdaily** [comment]
thanks 
good luck for you

**Comment by jeuxdaily** [comment]
hello again

How do I add Adjust ANE in a project instead of milkmanplugins.

I'm working on adobe animate cc

thanks

**Comment by ventr1x** [comment]
What is the current state of this. The provided ane does not seem to work. It crashes the app on first startup (interstingly only when installing from store):

> 08-29 08:39:12.860 19180 19180 D AndroidRuntime: Shutting down VM
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: FATAL EXCEPTION: main
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: Process: packageName, PID: 19180
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: java.lang.RuntimeException: Unable to instantiate receiver com.adjust.sdk.AdjustReferrerReceiver: java.lang.ClassNotFoundException: Didn't find class "com.adjust.sdk.AdjustReferrerReceiver" on path: DexPathList[[zip file "/data/app/packageName-e_FOC5uIoCogWMdXC7an9g==/base.apk"],nativeLibraryDirectories=[/data/app/packageName-e_FOC5uIoCogWMdXC7an9g==/lib/arm64, /data/app/packageName-e_FOC5uIoCogWMdXC7an9g==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64, /product/lib64]]
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3785)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at android.app.ActivityThread.-wrap18(Unknown Source:0)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2082)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:108)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:166)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7529)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: Caused by: java.lang.ClassNotFoundException: Didn't find class "com.adjust.sdk.AdjustReferrerReceiver" on path: DexPathList[[zip file "/data/app/packageName-e_FOC5uIoCogWMdXC7an9g==/base.apk"],nativeLibraryDirectories=[/data/app/packageName-e_FOC5uIoCogWMdXC7an9g==/lib/arm64, /data/app/packageName-e_FOC5uIoCogWMdXC7an9g==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64, /product/lib64]]
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:93)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3780)
> 08-29 08:39:12.861 19180 19180 E AndroidRuntime: 	... 8 more
> 08-29 08:39:12.864  1179  1231 I ActivityManager: Displayed packageName/.AppEntry: +1s942ms (total +16m17s311ms)

**Comment by uerceg** [comment]
Hi @ventr1x.

Hm. Interesting. I am not sure where does this originate from since our native Android SDK JAR which is packed inside of ANE I attached in [here](https://github.com/adjust/adobe_air_sdk/issues/49#issuecomment-521235879) should contain that class.

@jeuxdaily Did you have a chance to check that ANE version out and does it work for you?

**Comment by rewb0rn** [comment]
@ventr1x We have not yet released our apps via Google Play. I will keep an eye on this, thanks for sharing. I will update when I have some clarity if it is happening for us as well.

**Comment by jeuxdaily** [comment]
Did you have a chance to check that ANE version out and does it work for you?

it's work but it's only support 3000 devices

my next question how can create multipl APKS files

Do you have a chance to check the version of ANE and does it work for you?

Thank you for your interest

**Comment by uerceg** [comment]
@rewb0rn Just to double check on this one - is it safe to assume that version we sent you to test works well? If yes, we'd announce it publicly as well.

**Comment by rewb0rn** [comment]
Hi, yes it seems that everything is working correctly, although we had to roll back the release due to a bug not related to your ANE. So the data we have is limited.

**Comment by mayainle** [comment]
Any news when this version will be released?

**Comment by uerceg** [comment]
Hey. Yes, we'll try to address this next week and hopefully publish updated SDK. Will keep you posted and thanks for checking!

**Comment by majamatic** [comment]
Hi, do you already know when the new version will be published?

**Comment by uerceg** [comment]
Hi @majamatic 

We'll try to push it today, worst case tomorrow. Will keep you posted in here.

Cheers

**Comment by majamatic** [comment]
Thank you :)

**Comment by uerceg** [comment]
Okay, we just released 4.18.0 update which should contain support for 64-bit Android apps: https://github.com/adjust/adobe_air_sdk/releases/tag/v4.18.0

Thank you all for comments, patience and advices. Closing this anthological issue. In case any issue with the update pops up on this topic, feel free to reopen it.

Cheers and thanks one more time! 🍺
